src/sdam/topology.ts

```diff
@@ -968,8 +964,20 @@ function processWaitQueue(topology: Topology) {
       continue;
     }
 
-    const selectedServerDescription = randomSelection(selectedDescriptions);
-    const selectedServer = topology.s.servers.get(selectedServerDescription.address);
+    let selectedServer;
+    if (selectedDescriptions.length === 1) {
+      selectedServer = topology.s.servers.get(selectedDescriptions[0].address);
+    } else {
```

Comment:
Is it worth adding a special case for 2 servers here? 2 servers can be selected and compared, 3 or more have to be shuffled, right?